covid19-etl/etl/cityofchicago.py

```diff
@@ -354,36 +367,44 @@ def files_to_submissions(self):
         start = time.time()
         latest_submitted_date = self.metadata_helper.get_latest_submitted_date_idph()
 
-        # The following condition is for the first entry in dataset, which is from date `03-01-2020`
-        if latest_submitted_date == None:
-            latest_submitted_date = datetime(2020, 3, 1)
-
         today = datetime.today().replace(hour=0, minute=0, second=0, microsecond=0)
 
-        if latest_submitted_date == today:
+        if latest_submitted_date != None and latest_submitted_date == today:
             print("Nothing to submit: today and latest submitted date are the same.")
             return
 
-        print(
-            f"Latest submitted date: {latest_submitted_date}. Getting data until date: {today}"
-        )
-
         summary_location_submitter_id = format_submitter_id(
             "summary_location",
             {"country": self.country, "state": self.state, "city": self.city},
         )
 
+        self.last_submission_identifier = self.metadata_helper.get_last_submission()
+
+        # The following condition is for the first entry in dataset, which is from date `2020-03-01`
+        if self.last_submission_identifier == None:
+            self.last_submission_identifier = "2020-03-01"
+
+        print(
+            f"Latest submitted date: {latest_submitted_date}. Getting Missing data from date: {self.last_submission_identifier} 00:00:00 until date: {today}"
+        )
+
         self.get_summary_location(summary_location_submitter_id)
         self.parse_cityofchicago_file(
-            latest_submitted_date.strftime("%Y-%m-%dT%H:%M:%S.%f"),
+            self.last_submission_identifier,
             today.strftime("%Y-%m-%dT%H:%M:%S.%f"),
             summary_location_submitter_id,
         )
         print("Done in {} secs".format(int(time.time() - start)))
 
     def submit_metadata(self):
-        # Submits the data in `self.summary_locations`, `self.summary_clinicals` and `self.summary_group_demographic` to Sheepdog.
+        # Submits the data in `self.last_submission_identifier`, `self.summary_locations`, `self.summary_clinicals` and `self.summary_group_demographic` to Sheepdog.
         print("Submitting data...")
+
+        print("Updating last submission identifier date for project")
+        self.metadata_helper.update_last_submission(
+            self.last_submission_identifier.split("T")[0]
+        )
```

Comment:
let's do this after the other submissions, not before: if the other submissions fail, next time the job runs it should start at the same `last_submission_identifier` as before

Reply:
okay